### PR TITLE
Hotfix/2302 2344 article metadata errors

### PR DESCRIPTION
**Pull request by amdomanska**
Fixed errors:

1. for empty abstract elements the Type Error is not raised
https://github.com/DOAJ/doajPM/issues/2344#event-3252405301

2. An empty form is returned after successful upload of article's metadata:
https://github.com/DOAJ/doajPM/issues/2402

If ANY error occurred (including duplicate errors) returned form is pre-filled with last metadata entered for the easy proof-reading. Is that correct behaviour?

I am not able to reproduce the error from p1: https://github.com/DOAJ/doajPM/issues/2402#issue-603246918

3. If admin uploads the article without upload_article_id (like via publisher's metadata form or API) it is treated as publishers upload) - even though not used in practical context it makes testing easier for admin accounts.

### CODE REVIEW

File: portality/formcontext/formcontext.py [hunk 1]
```diff
@@ -1885,6 +1884,11 @@ def __init__(self, source, form_data, user):
     def set_template(self):
         self.template = "publisher/metadata.html"
 
+    def render_template(self, **kwargs):
+        self._check_for_author_errors()
```
Comment (Steven-Eardley) [inline portality/formcontext/formcontext.py:1888]: you're not calling `_check_for_author_errors()` with kwargs, so will it find anything?

File: portality/view/publisher.py [hunk 2]
```diff
@@ -196,15 +196,17 @@ def metadata():
...
             try:
                 fc.finalise()
+                validated = True
             except ArticleMergeConflict:
                 Messages.flash(Messages.ARTICLE_METADATA_MERGE_CONFLICT)
             except DuplicateArticleException:
                 Messages.flash(Messages.ARTICLE_METADATA_UPDATE_CONFLICT)
 
-        return fc.render_template()
+        return fc.render_template(validated=validated)
```
Comment (Steven-Eardley) [inline portality/view/publisher.py:209]: oh this is rather cool!

File: portality/formcontext/formcontext.py (hunk 1 above)
Comment (amdomanska) [reply portality/formcontext/formcontext.py:1888]: Good point.

File: portality/view/publisher.py (hunk 2 above)
Comment (amdomanska) [reply portality/view/publisher.py:209]: is it? I'm glad :D

**Pull request closed without merging**